torch/_dynamo/utils.py

```diff
@@ -3419,20 +3419,18 @@ def flatten_graph_inputs(gm: torch.fx.GraphModule, inputs, compile_gm):
         if node.op == "placeholder" and node.meta.get("steal_arg", False)
     ]
 
-    if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+    if is_compiled_autograd_gm(gm):
```

Comment:
This is more of an "educate me about what the code used to do situation". Let's suppose I graph break in the middle of a compiled autograd region. When I resume compilation on the resumption function, I imagine I wouldn't have a compiled autograd gm anymore, right? So is it OK to not go into this condition? Maybe the argument is that the boxed arguments only occur on entry to the very top of the compiled autograd graph? But then when I graph break and resume, I will have a lot of intermediate stack entries that will get fed in as non-boxed inputs, will these get promptly deallocated?

Reply:
Like maybe something like https://github.com/pytorch/pytorch/issues/122512 ?

Reply:
> When I resume compilation on the resumption function, I imagine I wouldn't have a compiled autograd gm anymore, right? So is it OK to not go into this condition?

The resume function takes non-boxed inputs, so we don't call this `flatten_graph_inputs` altogether. Even if we did change the resume function to take boxed inputs, it shouldn't be possible to have a graph break happen before we unpack the boxed inputs (first nodes after placeholder in the graph)

> when I graph break and resume, I will have a lot of intermediate stack entries that will get fed in as non-boxed inputs, will these get promptly deallocated?

No these won't deallocate until the end of that graph